sabnzbd/deobfuscate_filenames.py

```diff
@@ -225,90 +233,151 @@ def deobfuscate(nzo, filelist: List[str], usefulname: str):
     new_name = get_unique_filename("%s%s" % (os.path.join(path, usefulname), get_ext(biggest_file)))
     logging.info("Deobfuscate renaming %s to %s", biggest_file, new_name)
-    renamer(biggest_file, new_name)
+    filtered_filelist.remove(biggest_file)
+    filtered_filelist.append(renamer(biggest_file, new_name))
     nr_files_renamed += 1
 
     # Now find other files with the same basename in filelist, and rename them in the same way:
     basedirfile = get_basename(biggest_file)  # something like "/home/this/myiso"
-    for otherfile in filelist:
+    for otherfile in filtered_filelist[:]:
         if otherfile.startswith(basedirfile) and os.path.isfile(otherfile):
             # yes, same basedirfile, only different ending
             remaining_ending = otherfile.replace(basedirfile, "")  # might be long ext, like ".dut.srt" or "-sample.iso"
             new_name = get_unique_filename("%s%s" % (os.path.join(path, usefulname), remaining_ending))
             logging.info("Deobfuscate renaming %s to %s", otherfile, new_name)
-            # Rename and make sure the new filename is unique
-            renamer(otherfile, new_name)
+            filtered_filelist.remove(otherfile)
+            filtered_filelist.append(renamer(otherfile, new_name))
             nr_files_renamed += 1
 
     if nr_files_renamed:
         nzo.set_unpack_info("Deobfuscate", T("Deobfuscate renamed %d file(s)") % nr_files_renamed)
+
+    return filtered_filelist
+
+
+def without_extension(fullpathfilename):
+    """Returns full file path, without extension
+    So '/some/dir/somefile.bin' results in '/some/dir/somefile'"""
+    return os.path.splitext(fullpathfilename)[0]
+
+
+def deobfuscate_subtitles(input):
+    """
+    input can be an existing directory or a list of existing filenames
+
+    Find .srt subtitle files, and rename to match the biggest file (if there is a clearly biggest file)
+
+    Some_Big_File_2024.mp4      # biggest file
+    Some_Big_File_2024.srt      # no renaming wanted
+    Some_Big_File_2024.ger.srt  # no renaming wanted
+    14_English.srt              # to be renamed
+    dut.srt                     # to be renamed
+    Something.else.txt          # no renaming wanted, because no .srt
+
+    will result in
+
+    Some_Big_File_2024.mp4
+    Some_Big_File_2024.srt
+    Some_Big_File_2024.ger.srt
+    Some_Big_File_2024.14.English.srt   # renamed by prepending base name (and replacing _)
+    Some_Big_File_2024.dut.srt          # renamed by prepending base name
+    Something.else.txt
+
+    """
+
+    # if directory, create list from that directory
+    if type(input) == str and os.path.isdir(input):
+        all_files = [os.path.join(root, file) for root, dirs, files in os.walk(input) for file in files]
+    elif type(input) == list:
+        # already a list, so nothing to do
+        all_files = input
+    else:
+        logging.debug(f"{input} no directory and no list")
+        return None
```

Comment:
This whole section seems unnecessary, since we always use a list as input.
Just assume that the input is a list and then modify the tests to input a list instead of a directory.